src/cdp/msg.zig

```diff
@@ -0,0 +1,214 @@
+// Copyright (C) 2023-2024  Lightpanda (Selecy SAS)
+//
+// Francis Bouvier <francis@lightpanda.io>
+// Pierre Tachoire <pierre@lightpanda.io>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+const std = @import("std");
+
+// Parse incoming protocol message in json format.
+pub const IncomingMessage = struct {
+    scanner: std.json.Scanner,
+    json: []const u8,
+
+    obj_begin: bool = false,
+    obj_end: bool = false,
```

Comment:
I'm not sure it's mandatory to keep those information in the struct.
I think having them as variables at the beginning of the `scanUntil` method should be enough.

Reply:
It seems you are correct. 
I was thinking the state between two calls must be stored, but it works correctly with local variables.

Reply:
Hum, after more tests, I'm not sure if we can.

Reply:
I confirm we must keep the state.